libbpf-cargo/src/gen/mod.rs

```diff
@@ -208,6 +212,9 @@ pub(crate) fn canonicalize_internal_map_name(s: &str) -> Option<InternalMapType>
         // The `*.link` extension really only sets an additional flag in lower
         // layers. For our intents and purposes both can be treated similarly.
         Some(InternalMapType::StructOps)
+    } else if let Some(idx) = s.find(".data.") {
```

Comment:
can you please also add ".rodata.*" and ".bss.*" for completeness?

also see https://github.com/libbpf/libbpf/blob/master/src/libbpf.c#L1789-L1794, I don't think you need this complicated logic, just check for ".data." prefix.

Reply:
Thanks for the pointer. All done now.